frontend/pages/create-edit-user.js

```diff
@@ -0,0 +1,424 @@
+import { useState, useEffect, useRef, useReducer } from 'react';
+import {
+    client as lensClient,
+    createProfile,
+    getProfileByAddress,
+    getProfileByHandle,
+    createSetProfileWithMetadata,
+} from '../clientApi';
+import connectContract from '../utils/connectContract';
+// Components
+import Form from '../components/UI/Form';
+import Input from '../components/UI/Input';
+
+const defaultFormValues = {
+        name: '',
+        nameIsValid: null,
+        email: '',
+        emailIsValid: null,
+        username: '',
+        usernameIsValid: null,
+        website: '',
+        websiteIsValid: null,
+        bio: '',
+        bioIsValid: null,
+        links: {
+            twitter: { value: '', isValid: null },
+            lenster: { value: '', isValid: null },
+            instagram: { value: '', isValid: null },
+            discord: { value: '', isValid: null },
+            github: { value: '', isValid: null },
+        },
+    },
+    userFormReducer = (state, action) => {
+        const validators = {
+            email: /[^@]+@[A-Za-z0-9-]+\.[A-Za-z]+/,
+            username: { grep: /[A-Za-z0-9-]+/, len: 2 },
+            website: { grep: /[A-Za-z0-9-]+\.[A-Za-z]+/, len: 5 },
+            bio: { min: 1, max: 120 },
+            twitter: /https:\/\/twitter\.com\/[A-Za-z0-9]+/,
+            lenster: /https:\/\/lenster\.xyz\/u\/[A-Za-z0-9]+/,
+            instagram: /https:\/\/www.instagram.com\/[A-Za-z0-9]+/,
+            discord: /[A-Za-z0-9]+#\d{4}/,
+            github: /https:\/\/github\.com\/[A-Za-z0-9]+/,
+        };
```

Comment:
I like the idea to be more consistent across our code!

Reply:
Committed at 09f178c and updated at a258db0

Reply:
I was wondering if getting some of this logic to the utils folder is a good idea, or where do you think it could be taken to make a simpler file for the form?